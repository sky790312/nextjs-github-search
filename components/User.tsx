/* eslint-disable @typescript-eslint/camelcase */
import styled from 'styled-components'
import { GithubUserSchema } from '@/api/schema'
import Link from 'next/link'

export interface Props {
  user: GithubUserSchema
}

export const User: React.FC<Props> = ({ user }) => {
  return (
    <Link href={`/user?userName=${user.login}`}>
      <UserWrapper>
        <StyledUserAvatar src={user.avatar_url} />
        <div>
          <h3>{user.login}</h3>
        </div>
      </UserWrapper>
    </Link>
  )
}

const UserWrapper = styled.div`
  margin: 10px;
  padding: 20px;
  cursor: pointer;
  background-color: ${props => props.theme.colors.gray};

  &:hover {
    opacity: 0.7;
  }

  @media (min-width: 1140px) {
    display: flex;
    align-items: center;
  }
`

const StyledUserAvatar = styled.img`
  width: 100%;

  @media (min-width: 1140px) {
    width: 200px;
    margin-right: 20px;
  }
`
