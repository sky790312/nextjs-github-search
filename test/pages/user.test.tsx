import React from 'react'
import { render, withProviders } from '@/test/testUtils'
import User from '@/pages/user'

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '',
      pathname: '',
      query: '',
      asPath: '',
    }
  },
}))

describe('User page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(withProviders(<User />), {})
    expect(asFragment()).toMatchSnapshot()
  })
})
