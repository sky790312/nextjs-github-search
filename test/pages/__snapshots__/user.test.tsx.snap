// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="GlobalStyles__Container-sc-1m5drgn-0 frTxMm"
  >
    <h1 />
    <div
      class="GlobalStyles__ListContainer-sc-1m5drgn-2 lDTWr"
    />
    <div
      class="GlobalStyles__TextCenterContainer-sc-1m5drgn-1 LoadingSection__LoadingSectionWrapper-ekcqrm-0 kwExyG dzDQFW"
    >
      <button
        class="LoadingSection__Button-ekcqrm-1 iPqPCv"
      >
        Load More
      </button>
    </div>
  </div>
</DocumentFragment>
`;
